### PR TITLE
Feature/enhance logger

### LINKED ISSUE #68: Logger enhancement
#### Description
Optimize logger function
#### Expected Behavior
```
2020-09-09 16:58:02
Filename : test2.gr
Vertices : 2
Total Edges : 4
Max Edge: x
Min Edge: x
Directed: True/False
Signed: True/False
Multigraph: True/False
Self Loop: True/False
Weighted: True/False
Max Weight: x
Min Weight: x
Elapsed Time : 00 days, 00 hour, 00 minutes, 20 seconds
```
#### Actual Behavior
```
2020-09-09 16:58:02.103878
Filename : test2.gr
Vertices : 2
Edges : 4
Elapsed Time : 00 days, 00 hour, 00 minutes, 20 seconds
```
#### Operating System
Windows 10
#### Python Version
3.5.2
#### Pyrgg Version
0.8

### PR DESCRIPTION
**Pull request by ahmadsalimi**
#### Reference Issues/PRs
#68 

#### What does this implement/fix? Explain your changes.
Enhance the logger function by adding the following parameters to the log file:
- Max Edge
- Min Edge
- Directed
- Signed
- Multigraph
- Self Loop
- Weighted
- Max Weight
- Min Weight

Also change the time format to `%Y-%m-%d %H:%M:%S`.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,8 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+### Changed
+- `logger` function enhanced with additional parameters such as `max_edge`, `min_edge`, `directed`, `signed`, `multigraph`, `self_loop`, `weighted`, `max_weight`, `min_weight`.
```
Comment (sadrasabouri) [inline CHANGELOG.md:9]: Could you please make these change logs atomic?
something like:
```
### Added
- 9 new parameters to `logger`:
    + `max_edge`
    + `min_edge`
    ...
```

**Review by sadrasabouri — APPROVED** [review]
@sepandhaghighi I'm done, can we add a test for printing generated file from `logger` ignoring first line (date and time)?

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,21 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
...
+    - `directed`
+    - `signed`
+    - `multigraph`
+    - `self_loop`
+    - `weighted`
+    - `max_weight`
+    - `min_weight`
+
+### Changed
+- Time format in the `logger` changed to `%Y-%m-%d %H:%M:%S`. e.g. `2020-10-12 20:25:47`
```
Comment (sepandhaghighi) [inline CHANGELOG.md:21]: Please remove example section

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,6 +5,21 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+### Added
+- **9** new parameters to the `logger` function as below and log them too:
```
Comment (sepandhaghighi) [inline CHANGELOG.md:9]: Added section is for new features, format, or file.
And also logger is an inner function, only add a general sentence in changed section, for example :
- `logger` function modified

**Review by sepandhaghighi — CHANGES_REQUESTED** [review]
@ahmadsalimi Thanks for your effort 💯 

1. Please solve those minor issues in `CHANGELOG.md`
2. Take a look at @sadrasabouri comment about `logger` function test cases

**Comment by sepandhaghighi** [comment]
> @sepandhaghighi I'm done, can we add a test for printing generated file from `logger` ignoring first line (date and time)?

Good idea  💯

File: CHANGELOG.md (hunk 2 above)
Comment (ahmadsalimi) [reply CHANGELOG.md]: Fixed

File: CHANGELOG.md (hunk 3 above)
Comment (ahmadsalimi) [reply CHANGELOG.md]: Fixed

File: pyrgg/functions.py [hunk 4]
```diff
@@ -120,28 +120,68 @@ def filesize(fileaddr):  # pragma: no cover
...
+                "Min Edge : " + str(min_edge) + "\n" +
+                "Directed : " + str(bool(directed)) + "\n" +
+                "Signed : " + str(bool(signed)) + "\n" +
+                "Multigraph : " + str(bool(multigraph)) + "\n" +
+                "Self Loop : " + str(bool(self_loop)) + "\n" +
+                "Weighted : " + str(bool(weighted)) + "\n" +
+                "Max Weight : " + str(max_weight) + "\n" +
+                "Min Weight : " + str(min_weight) + "\n" +
+                "Elapsed Time : " + elapsed_time + "\n" +
+                "-------------------------------\n")
```
Comment (ahmadsalimi) [inline pyrgg/functions.py:184]: There was a problem that I fixed. Suppose `file_name` parameter isn't `string`. So, the line
```python
file.write("Filename : " + file_name + "\n")
```
raises an error. but the first line of log, which is
```python
file.write(datetime.datetime.now().strftime('%Y-%m-%d %H:%M:%S') + "\n")
```
is written in the file. So, I'm using just one `file.write()` function call to fix this error.

@sepandhaghighi @sadrasabouri

Comment (ahmadsalimi) [inline pyrgg/functions.py:184]: There was a problem that I fixed. Suppose the `file_name` parameter isn't a `string`. So, the line
```python
file.write("Filename : " + file_name + "\n")
```
raises an error. but the first line of log, which is
```python
file.write(datetime.datetime.now().strftime('%Y-%m-%d %H:%M:%S') + "\n")
```
is written in the file. So, I decided to use just one `file.write()` function call to fix this error.

@sepandhaghighi @sadrasabouri

File: pyrgg/test.py [hunk 5]
```diff
@@ -39,7 +39,22 @@
...
+Max Edge : 1000
+Min Edge : 10
+Directed : True
+Signed : False
+Multigraph : False
+Self Loop : True
+Weighted : True
+Max Weight : 20
+Min Weight : 1
+Elapsed Time : 2min
```
Comment (ahmadsalimi) [inline pyrgg/test.py:57]: This part is added to test the output of the `logger` function.

@sepandhaghighi @sadrasabouri

**Review by sepandhaghighi — APPROVED** [review]

**Comment by sepandhaghighi** [comment]
@ahmadsalimi Well done 🥇 
In the `logger` function you assumed flags as `0` and `1` so this should be considered in #69.

**Pull request merged**